core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -194,7 +196,12 @@ public String getLocation(@Nullable String routing) {
         if (routing != null) {
             location.append(routingStart).append(routing);
         }
-        return location.toString();
+        try {
+            URI uri = new URI(location.toString());
+            return uri.toASCIIString();
+        } catch (URISyntaxException ex) {
```

Comment:
I think using an exception for flow control should be avoided here. Is there another way?